google/cloud/aiplatform/datasets.py

```diff
@@ -15,6 +15,358 @@
+            data_item_labels (Optional[Dict]):
+                Labels that will be applied to newly imported DataItems. If
+                an identical DataItem as one being imported already exists
+                in the Dataset, then these labels will be appended to these
+                of the already existing one, and if labels with identical
+                key is imported before, the old label value will be
+                overwritten. If two DataItems are identical in the same
+                import data operation, the labels will be combined and if
+                key collision happens in this case, one of the values will
+                be picked randomly. Two DataItems are considered identical
+                if their content bytes are identical (e.g. image bytes or
+                pdf bytes). These labels will be overridden by Annotation
+                labels specified inside index file refenced by
+                [import_schema_uri][google.cloud.aiplatform.v1beta1.ImportDataConfig.import_schema_uri],
+                e.g. jsonl file.
+        Returns:
+            dataset (Dataset):
+                Instantiated representation of the managed dataset resource.
+        """
+
+        import_lro = self._import(
+            source=gcs_source,
+            import_schema_uri=import_schema_uri,
+            data_items_labels=data_items_labels,
+        )
+
+        import_lro.result()  # An empty response upon successful import
+
+        return self
+
+    def export_data(self, output_dir: str) -> Optional[operation.Operation]:
+        """Exports data to output dir to GCS.
+
+        Args:
+            output_dir (str):
+                Required. The Google Cloud Storage location where the output is to
+                be written to. In the given directory a new directory will be
+                created with name:
+                ``export-data-<dataset-display-name>-<timestamp-of-export-call>``
+                where timestamp is in YYYYMMDDHHMMSS format. All export
+                output will be written into that directory. Inside that
+                directory, annotations with the same schema will be grouped
+                into sub directories which are named with the corresponding
+                annotations' schema title. Inside these sub directories, a
+                schema.yaml will be created to describe the output format.
+
+                If the uri doesn't end with '/', a '/' will be automatically
+                appended. The directory is created if it doesn't exist.
+
+        Returns:
+            operation (Operation):
+                An object representing a long-running operation.
+        """
+
+        export_data_config = ExportDataConfig(
+            gcs_destination=GcsDestination(output_uri_prefix=output_dir)
+        )
+
+        export_lro = self.api_client.export_data(
+            name=self.resource_name, export_config=export_data_config
+        )
+
+        return export_lro
```

Comment:
To be symmetric with import_data, I think we should block here and return the output_dir uri upon completion. We should add async support for both as a single work item.